sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredentialOptions can be used when providing additional credential information, such as a client secret.
+// Also use these options to modify the default pipeline behavior through the TokenCredentialOptions.
+type InteractiveBrowserCredentialOptions struct {
+	ClientSecret *string                 // Gets the client secret that was generated for the App Registration used to authenticate the client. Only applies for web apps.
+	Options      *TokenCredentialOptions // Options allow to configure the management of the requests sent to Azure Active Directory.
+}
+
+// InteractiveBrowserCredential enables authentication to Azure Active Directory using an interactive browser to log in.
+type InteractiveBrowserCredential struct {
+	client       *aadIdentityClient
+	tenantID     string  // Gets the Azure Active Directory tenant (directory) ID of the service principal
+	clientID     string  // Gets the client (application) ID of the service principal
+	clientSecret *string // Gets the client secret that was generated for the App Registration used to authenticate the client.
+}
+
+// NewInteractiveBrowserCredential constructs a new InteractiveBrowserCredential with the details needed to authenticate against Azure Active Directory through an interactive browser window.
+// tenantID: The Azure Active Directory tenant (directory) ID of the service principal.
+// clientID: The client (application) ID of the service principal.
+// clientSecret: Gets the client secret that was generated for the App Registration used to authenticate the client.
+// options: allow to configure the management of the requests sent to Azure Active Directory.
+func NewInteractiveBrowserCredential(tenantID string, clientID string, options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
```

Comment:
`tenantId` and `clientId` should be moved to `options` as well, the are not required. `tenantId` should default to organizations, and `clientId` should default to the developer sign-on client id.

Reply:
Done